dependency/catalog_node.go

```diff
@@ -152,5 +154,5 @@ type NodeServiceList []*NodeService
 func (s NodeServiceList) Len() int      { return len(s) }
 func (s NodeServiceList) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s NodeServiceList) Less(i, j int) bool {
-	return s[i].Service <= s[j].Service
```

Comment:
Seeing as ID defaults to Service if not provided when said service was registered I don't think that ordering by service then falling back to ID gives us anything as, for the most part, it will be the same value. However i'm not too fussed either way as long as the the ordering is the same between template renderings. :grin: